### PR TITLE
emailService Js -> ts

### PR DESCRIPTION
**Pull request by niccolopaganini**
- emailService Js -> ts
- modified files surrounding emailService

### CODE REVIEW

**Comment by niccolopaganini** [comment]
Manual Testing done. Process:
- ran `npm run serve` (after activation)
- connected the phone to Chrome's Developer tools and kept an eye out for issues.
- Then, I used my address (nseptank@nrel.gov) to send it to my personal email address (write2gamenugget@gmail.com) to make sure (I changed to Shankari's email address )
- I then went to my personal email address to check the email was received successfully and then the file was present.

**Comment by shankari** [comment]
@niccolopaganini as we discussed, please add a video of the testing done before requesting review.

**Comment by niccolopaganini** [comment]
This is a video of the devapp
https://github.com/e-mission/e-mission-phone/assets/69108657/374d36c3-0ffc-4043-bb5a-bcc472b93196

**Comment by niccolopaganini** [comment]
Email received (in my work email address) 

image

File: www/index.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ import './js/main.js';
 import './js/diary.js';
 import './js/diary/services.js';
 import './js/survey/enketo/answer.js';
-import './js/control/emailService.js';
+//import './js/survey/enketo/enketo-trip-button.js';
+//import './js/survey/enketo/enketo-add-note-button.js';
```
Comment (JGreenlee) [inline www/index.js:18]: Remove these lines please

File: www/js/control/emailService.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from 'react';
```
Comment (JGreenlee) [inline www/js/control/emailService.ts:1]:
```suggestion
```

File: www/js/control/emailService.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import i18next from 'i18next';
+import { logInfo, logDebug, displayError } from '../plugin/logger';
```
Comment (JGreenlee) [inline www/js/control/emailService.ts:3]:
```suggestion
import { logInfo } from '../plugin/logger';
```

File: www/js/control/emailService.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return;
+  }
+
+  if (window['ionic'].Platform.isAndroid()) {
+    parentDir = 'app://databases';
+  }
+
+  if (window['ionic'].Platform.isIOS()) {
+    alert(i18next.t('email-service.email-account-mail-app'));
+    console.log(window['cordova'].file.dataDirectory);
```
Comment (JGreenlee) [inline www/js/control/emailService.ts:27]:
```suggestion
    logDebug(window['cordova'].file.dataDirectory);
```

File: www/js/control/emailService.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    subject: i18next.t('email-service.email-log.subject-logs'),
+    body: i18next.t('email-service.email-log.body-please-fill-in-what-is-wrong'),
+  };
+
+  window['cordova'].plugins['email'].open(emailData, () => {
+    logInfo(
+      'Email app closed while sending, ' +
+        JSON.stringify(emailData) +
+        ' not sure if we should do anything',
+    );
```
Comment (JGreenlee) [inline www/js/control/emailService.ts:54]:
```suggestion
    logWarn(`Email app closed while sending, 
      emailData = ${JSON.stringify(emailData)}`);
```

File: www/index.js (hunk 1 above)
Comment (JGreenlee) [inline www/index.js:18]:
```suggestion
```

File: www/js/control/emailService.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    window['cordova'].plugins['email'].hasAccount((hasAct) => {
+      resolve(hasAct);
+    });
+  });
+}
+
+export async function sendEmail(database: string) {
+  let parentDir = 'unknown';
+
+  if (window['ionic'].Platform.isIOS() && !(await hasAccount())) {
```
Comment (JGreenlee) [inline www/js/control/emailService.ts:16]:
```suggestion
  if (window['cordova'].platformId == 'ios' && !(await hasAccount())) {
```

File: www/js/control/emailService.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    alert(i18next.t('email-service.email-account-not-configured'));
+    return;
+  }
+
+  if (window['ionic'].Platform.isAndroid()) {
```
Comment (JGreenlee) [inline www/js/control/emailService.ts:21]:
```suggestion
  if (window['cordova'].platformId) == 'android') {
```

File: www/js/control/emailService.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    parentDir = 'app://databases';
+  }
+
+  if (window['ionic'].Platform.isIOS()) {
```
Comment (JGreenlee) [inline www/js/control/emailService.ts:25]:
```suggestion
  if (window['cordova'].platformId) == 'ios') {
```

**Review by JGreenlee — APPROVED** [review]

**Comment by shankari** [comment]
@JGreenlee @niccolopaganini I am merging this right now in the interests of progress, but I would like to see tests for this show up fairly soon. Merging but moving into the "cleanup" column for now.

**Comment by JGreenlee** [comment]
@shankari I didn't ask for unit tests on this PR because I didn't think it would be possible to test the functionality of sending an email (which is basically all this service does)

We could mock the email plugin but then we're just testing the mock. Is there something else here that you see value to testing?

**Comment by shankari** [comment]
@JGreenlee I agree that mocking is not helpful. However, it seems like it should be possible to use Appium to test that the email client is launched successfully to at least check for issues like https://github.com/katzer/cordova-plugin-email-composer/issues

wrt checking whether the email is actually sent, we just need a service that lets you sent throwaway emails to it and then check their receipt via API. So something like https://testmail.app/

**Comment by JGreenlee** [comment]
> it seems like it should be possible to use Appium to test that the email client is launched successfully

Once we have Appium integrated, this might be possible in an Android emulator – but I am pretty sure it's not going to be achievable on the iOS Simulator because it doesn't support email at all.

Even so, I'm not sure if Appium can detect an external app launch. I think Appium attaches itself to the app it's targeted at and I'm not sure if it can detect anything from other apps.

> wrt checking whether the email is actually sent

This would be a function of the email client app (e.g. Gmail); why would we have to test an app that isn't ours? Isn't that beyond the scope of this project?

**Comment by shankari** [comment]
> Once we have Appium integrated, this might be possible in an Android emulator – but I am pretty sure it's not going to be achievable on the iOS Simulator because it doesn't support email at all.

Correct, it would only be in an android emulator.

> Even so, I'm not sure if Appium can detect an external app launch. I think Appium attaches itself to the app it's targeted at and I'm not sure if it can detect anything from other apps.

It would be interesting to know that. If not, I guess we must maintain a (hopefully short) list of tests that must be performed manually and add this there.

> This would be a function of the email client app (e.g. Gmail); why would we have to test an app that isn't ours? Isn't that beyond the scope of this project?

Well, we invoke gmail, and it would be good to know if we do so correctly. For example, if a merge conflict resulted in the email address being munged (e.g. foobar@goobar.commereefe), we would want to know. Similarly, if there is an error in exporting the logs, we would want to know. It would help us find issues like https://github.com/e-mission/e-mission-docs/issues/994

FYI, another email checking service:
https://mailtrap.io/blog/selenium-email-verification/

**Pull request merged**